java/src/main/java/com/logicalclocks/base/FeatureStoreBase.java

```diff
@@ -0,0 +1,231 @@
+/*
+ *  Copyright (c) 2020-2022. Hopsworks AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+package com.logicalclocks.base;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.logicalclocks.base.constructor.QueryBase;
+import com.logicalclocks.base.metadata.FeatureGroupApi;
+import com.logicalclocks.base.metadata.StorageConnectorApi;
+import com.logicalclocks.base.metadata.TrainingDatasetApi;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+
+import java.io.IOException;
+import java.util.List;
+
+public abstract class FeatureStoreBase<T extends FeatureStoreBase, T2 extends QueryBase> {
```

Comment:
```suggestion
public abstract class FeatureStoreBase<T2 extends QueryBase> {
```
as far a I know T is not used